source/PlanetPanel.cpp

```diff
@@ -108,19 +108,25 @@ void PlanetPanel::Draw()
 
 	if(planet.CanUseServices())
 	{
-		if(planet.IsInhabited())
-		{
+		const auto &port = planet.GetPort();
+
+		if(port.HasService(Port::ServicesType::Bank))
 			info.SetCondition("has bank");
-			if(flagship)
-			{
-				info.SetCondition("is inhabited");
-				if(system.HasTrade())
-					info.SetCondition("has trade");
-			}
+		if(flagship)
+		{
+			if(port.HasService(Port::ServicesType::JobBoard))
+				info.SetCondition("has job board");
+			if(port.HasService(Port::ServicesType::HireCrew))
+				info.SetCondition("can hire crew");
+			if(port.HasService(Port::ServicesType::Trading) && system.HasTrade())
+				info.SetCondition("has trade");
 		}
 
-		if(flagship && planet.HasSpaceport())
-			info.SetCondition("has spaceport");
+		if(flagship && planet.HasPort())
+		{
+			info.SetCondition("has port");
+			info.SetString("port name", port.Name());
+		}
```

Comment:
It feels a bit weird to me that we're checking whether the port from the planet has specific services at the top, but then at the bottom we're asking "Well does the planet even actually have a defined port?" I feel like that's the first thing we should be doing, and everything else should only follow if that's true. So something like this instead.
```
if(planet.HasPort())
{
	Port port = planet.GetPort();
	if(port.HasBank())
		info.SetCondition("has bank");
	if(flagship)
	{
		info.SetCondition("has port")
		... Then check all the other services and stuff ...
	}
}
```

Reply:
Notably, Planet::HasPort checks Port::IsValid, which just checks that the port has a name. So if the intention is that a port could have services but no name/description, then HasPort and IsValid should probably have their names changed, and/or a comment could describe that allowed possibility.